### ISSUE TITLE
Pxssh broken options in current pip version

### ISSUE THREAD

**Issue by Alexandertyler** [pexpect/pexpect#179]
Hi. I've been working on pxssh for vm testing. I've hit some pretty serious issues with the way pxssh deals with ssh options on the current version released. It seems like it might be fixed in the dev version but I wanted to post this here in case anyone has this issue and the workaround that I discovered for it. Let me explain the problem first:
>I attempted to pass the pxssh shell `StrictHostKeyChecking=no +UserKnownHostsFile /dev/null` 
>In the latest version of the pexpect docs, it gives a couple of different steps to do this, none of which work. I dug into the source file that pip and even the latest tar ball installs and discovered some key differences between the docs and the latest version on github. It breaks any option for users to set ssh options that will be invoked when the command is called. You can see why this would be a big deal. Apart from the code not matching the documents it prevents users from making any sort of distributable program with pxssh because if the host key doesn't match, like when using a VM with a new drive, the programs errors out on an EOF. 

The work-around:

>When I explored the source file that pip installed, I found 

```python
self.SSH_OPTS = ("-o'RSAAuthentication=no'"
                   + " -o 'PubkeyAuthentication=no'")
# Disabling host key checking, makes you vulnerable to MITM attacks.
#                + " -o 'StrictHostKeyChecking=no'"
#                + " -o 'UserKnownHostsFile /dev/null' ")
```

>At some point there was a framework in place for SSH_OPTS. We can't just set this field after initialization though. In login(), the ssh_options to be passed on are set in the chunk 

```python
ssh_options = ''
        if quiet:
            ssh_options = ssh_options + ' -q'
        if not check_local_ip:
            ssh_options = ssh_options + " -o'NoHostAuthenticationForLocalhost=yes'"
        if self.force_password:
            ssh_options = ssh_options + ' ' + self.SSH_OPTS
        if port is not None:
            ssh_options = ssh_options + ' -p %s'%(str(port))
        if ssh_key is not None:
            try:
                os.path.isfile(ssh_key)
            except:
                raise ExceptionPxssh('private ssh key does not exist')
            ssh_options = ssh_options + ' -i %s' % (ssh_key)
        cmd = "ssh %s -l %s %s" % (ssh_options, username, server)
```

>On the force_password check, it adds in the self.SSH_OPTS to the command. So to perform the workaround, after initialization of the shell object, set SSH_OPTS to be whatever you need (in my case  shell.SSH_OPTS = (" -o'StrictHostKeyChecking=no'" + " -o'UserKnownHostsFile /dev/null'")) and then set the flag force_password = True. Everything else should work as intended. 

DISCLAIMER: Using this workaround is probably not something that the developers intended. When pexpect gets updated with the latest code on github it will break this scheme. But that's not a bad thing. It will make setting options easier than this. To fix your code after the update you will have to remove the changes and instead follow the online documentation for setting options.

**Comment by Alexandertyler** [comment]
FYI, the pexpect version hosted right now is 3.3

**Comment by jquast** [comment]
Same problem as https://github.com/pexpect/pexpect/issues/104#issuecomment-141860482

Further confusing, pexpect documentation for upcoming 4.0 release is published as release 3.3 on readthedocs.org, address by PR #257 @takluyver please ensure to synchronize your readthedocs.org administration label in coordination with pypi package upload so that the upcoming pexpect release is not reflected, or agree to use separate 'develop' branch for future.

**Comment by jquast** [comment]
Also, duplicate of #235 for same cause (version management booboo)
